File: 백준/일곱-난쟁이.py
from sys import stdin

input = stdin.readline

arr = [int(input()) for _ in range(9)]
overflow = sum(arr) - 100

for i in range(8):
    if arr[i] > overflow:
        continue

    for j in range(i + 1, 9):
        if (arr[i] + arr[j] == overflow):
            arr = sorted(list(filter(lambda x: x != arr[i] and x != arr[j], arr)))
            break

    if (len(arr) == 7):
        break

print(*arr, sep='\n')

File: 백준/일곱-난쟁이.js
let input = require('fs')
  .readFileSync('/dev/stdin')
  .toString()
  .split('\n')
  .map((e) => +e.trim());
const overflow = input.reduce((acc, cur) => acc + cur) - 100;

for (let i = 0; i < 8; i++) {
  if (input[i] > overflow) continue;

  for (let j = i + 1; j < 9; j++) {
    if (input[i] + input[j] === overflow) {
      input = input.filter((e) => e !== input[i] && e !== input[j]).sort((a, b) => a - b);
      break;
    }
  }

  if (input.length === 7) break;
}

console.log(input.join('\n'));
